### PR TITLE
Image sharing react setup

### LINKED ISSUE #10: React Setup
#### As a developer I want to be able to integrate React into my rails application.

__Story__:

Your users are loving the image sharer application, and want to give their opinions on how certain features can be improved. Using React and MobX, we will enable users to submit feedback for the application.

We will practice the act of “slicing” stories into small, quick to test and deliver parts. For this issue, you will be asked to add footer text to the existing boilerplate code. There are existing TODO comments in the code.

In `app/assets/javascripts/image_sharing` you will find the starting code for this feedback form. The rundown of directories are:
- `components` holds all React components, potentially namespaced by more directories
- `services` holds the service objects you may need - some examples include api calls, asynchronous I/O, etc.
- `stores` holds MobX state stores (see references/documentation related to MobX)
- `utils` holds general purpose helper modules that can be shared across different parts of the client application
- `test` holds tests for all functional code in the application. It's directory structure usually mimics that of the source. You may need to create this directory if it doesn't already exist.

We've created a starting point for you in the form of a branch in your repo called `ImageSharingReactSetup`. The first step is to rebase the `ImageSharingReactSetup` branch on top of your updated master branch.

Happy coding!

__Acceptance criteria__:
- [x] The React client app loads properly and a user can access the feedback form
- [x] There is a footer at the bottom of the feedback form that reads: “Copyright: Appfolio Inc. Onboarding”
- [x] There is a unit test that checks that the footer text exists

__Discussion Topics__:
- [ ] What is the difference between stateful and stateless components in React?
- [ ] Why might it be good to consider stateless components?
- [ ] How might one set up a test environment to run for react?

__References__:
* https://reactjs.org/tutorial/tutorial.html
* https://mobx.js.org/getting-started.html
* https://airbnb.io/enzyme/
* https://mochajs.org/
* https://nodejs.org/api/assert.html
* https://www.chaijs.com/guide/

### PR DESCRIPTION
**Pull request by HangDu1995**
Closes #10 

---

The following is a checklist to remind you of the steps to take when opening a
new PR. The horizontal line above and everything below can be removed from the
pull request description.

- [ ] I have replaced `X` above with the associated issue number.
- [ ] I have changed the issue's label to "pull request".
- [ ] I have assigned the designated reviewer to this PR.
- [ ] I have set the appropriate label on this PR.
- [ ] I have written clear commit messages ([ref](http://chris.beams.io/posts/git-commit/)).

If needed, please review the document on the
[Image Sharing Project](https://sites.google.com/a/appfolio.com/eng/new-hire-resources/engineering-academy/ropes/image-sharing-project)
for information on how labels should be used.

### CODE REVIEW

File: app/views/feedbacks/new.html.erb [hunk 1]
```diff
@@ -1 +1,4 @@
 <div id="feedback-root"></div>
+<%= javascript_include_tag 'application' %>
```
Comment (choruk) [inline app/views/feedbacks/new.html.erb]: The `javascript_include_tag` helper is pulling from `app/assets/javascripts`, so this tag is saying to include the `app/assets/javascripts/application.js` file. You shouldn't need to include that file on this page.

File: app/views/feedbacks/new.html.erb [hunk 2]
```diff
@@ -1 +1,4 @@
...
+<%= javascript_pack_tag 'application' %>
```
Comment (choruk) [inline app/views/feedbacks/new.html.erb]: I think if you are going to move this `javascript_pack_tag` call into this file (which is a good idea to do), you should consider renaming it to something other than `application`. Within Rails, the convention behind the files named `application` is to have them be for things that apply to the whole app (e.g. `ApplicationController` should be the base controller class for all of the controllers in the app, `ApplicationRecord` should be the base model class for all of the models in the app, etc).

File: app/javascript/image_sharing/components/App.js [hunk 3]
```diff
@@ -1,21 +1,31 @@
...
-        <h1>React</h1>
-        /* Put your components here: Flash Message, Form, Footer */
+        <Header title="Tell us what you think" />
+
+        <form>
+          <label>
+              Comment:
+
+            <div>
+              <textarea type="text" name="comment" />
```
Comment (choruk) [inline app/javascript/image_sharing/components/App.js]: You definitely can use the actual HTML tags in here if you want, but I think you will have a much easier time developing if you use the components from reactstrap (e.g. [here are the reactstrap form components docs](https://reactstrap.github.io/components/form/)). Something to consider for your next issue.

File: app/views/feedbacks/new.html.erb (hunk 1 above)
Comment (HangDu1995) [reply app/views/feedbacks/new.html.erb]: Yeah you are right. Deleted it

File: app/views/feedbacks/new.html.erb (hunk 2 above)
Comment (HangDu1995) [reply app/views/feedbacks/new.html.erb]: Ok, I changed the package name to `feedback`

File: app/javascript/image_sharing/components/App.js (hunk 3 above)
Comment (HangDu1995) [reply app/javascript/image_sharing/components/App.js]: Yeah, I will do that in my next PR

**Pull request merged**